gengo/src/analysis/mod.rs

```diff
@@ -42,21 +65,8 @@ impl Analysis {
     }
 }
 
-pub struct Iter<'map>(IndexMapIter<'map, PathBuf, Entry>);
-
-impl<'map> Iterator for Iter<'map> {
-    type Item = (&'map PathBuf, &'map Entry);
-
-    fn next(&mut self) -> Option<Self::Item> {
-        self.0.next()
-    }
-}
-
-impl<'map> IntoIterator for &'map Analysis {
-    type Item = (&'map PathBuf, &'map Entry);
-    type IntoIter = Iter<'map>;
-
-    fn into_iter(self) -> Self::IntoIter {
-        self.iter()
+impl std::fmt::Debug for Analysis {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_map().entries(self.iter()).finish()
```

Comment:
Looks like this `Debug` implementation removed the type's identity. Do you think that it could be added back, so the output is
```
Analysis {
...
```

Reply:
Actually I tried but it's not possible without implementing parts of the Debug impl by hand. Since it's used for debugging only, I don't think it's a problem.
One could, of course, manually write `f.write_str("Analyis {")?;` and wrap it up with `f.write_str("}")`, but formatting will look different unless a lot of care is taken that wouldn't make sense for `insta` I think.